VocaDbWeb.Core/Startup.cs

```diff
@@ -50,7 +59,43 @@ public Startup(IConfiguration configuration)
 		// This method gets called by the runtime. Use this method to add services to the container.
 		public void ConfigureServices(IServiceCollection services)
 		{
-			services.AddControllersWithViews();
+			services
+				.AddControllersWithViews(options =>
+				{
+					// Code from: https://stackoverflow.com/questions/51411693/null-response-returns-a-204/60858295#60858295
+					// remove formatter that turns nulls into 204 - No Content responses
+					// this formatter breaks Angular's Http response JSON parsing
+					options.OutputFormatters.RemoveType<HttpNoContentOutputFormatter>();
+
+					options.Filters.Add<VoidAndTaskTo204NoContentFilter>();
+				})
+				.ConfigureApiBehaviorOptions(options =>
+				{
+					// Code from: https://docs.microsoft.com/en-us/aspnet/core/web-api/?view=aspnetcore-5.0#disable-inference-rules
+					// Disable inference rules. This prevents API controllers from returning an UnsupportedMediaTypeResult (415).
+					options.SuppressInferBindingSourcesForParameters = true;

```

Comment:
Is it okay to disable inference rules?

Reply:
Could you comment here why is this needed?

Reply:
But **why** is it returning unsupported media type?

Reply:
I don't know why. I found some possibly related issues:

- aspnet/Mvc#8116
- dotnet/aspnetcore#9932

Currently, [AlbumApiController.PostReview](https://github.com/VocaDB/vocadb/blob/817001d0fb8ebe121b0e4ab53dc8211941f5a3da/VocaDbWeb.Core/Controllers/Api/AlbumApiController.cs#L240) throws an `ObjectNotFoundException` since the album ID is passed not only to `id`, but also to `reviewContract.id`. We can solve this by adding the `[FromForm]` attribute to `AlbumReviewContract reviewContract`. Should we use this attribute instead of disabling inference rules? Originally, I tried to solve `Unsupported Media Type` with this attribute, I was not sure if this was the right way.

Reply:
I don't know why. I found some possibly related issues:

- aspnet/Mvc#8116
- dotnet/aspnetcore#9932

Currently, [AlbumApiController.PostReview](https://github.com/VocaDB/vocadb/blob/817001d0fb8ebe121b0e4ab53dc8211941f5a3da/VocaDbWeb.Core/Controllers/Api/AlbumApiController.cs#L240) throws an `ObjectNotFoundException` since the album ID is passed not only to `id`, but also to `reviewContract.id`. We can solve this by adding the `[FromForm]` attribute to `AlbumReviewContract reviewContract`. Should we use this attribute instead of disabling inference rules? Originally, I tried to solve `Unsupported Media Type` with this attribute, but I was not sure if this was the right way.

Reply:
Is the problem that it's binding the ID to wrong field? Adding the attribute sounds like more proper way of fixing it. Setting this global SuppressInferBindingSourcesForParameters sounds like a last resort workaround.

Reply:
Yes, I also tried `[FromBody]`, but in vain. Maybe related to [this SO question](https://stackoverflow.com/questions/49228906/415-unsupported-media-type-asp-net-core). Actually, I need to replace `[FromBody]` with `[FromForm]` in `UserApiController`.

Reply:
How about overriding `$.post` in `libs.js`?

```javascript
$.post = function (url, data, success, dataType) {
	return $.ajax({ url: url, type: "POST", dataType: dataType, contentType: "application/json", data: ko.toJSON(data), success: success });
};
```

Reply:
How about overriding `$.post` in `libs.js`?

```javascript
$.post = function (url, data, success, dataType) {
	return $.ajax({ url: url, type: "POST", dataType: dataType, contentType: "application/json", data: ko.toJSON(data), success: success });
};
```

Reply:
Thanks for the explanation!